requirements/specifications/openrpc-extensions/app-passthrough-apis.md

```diff
@@ -0,0 +1,533 @@
+
+If the platform method result `items` schema matches the `x-response` schema on the provider method then each provier value **MUST** be used as-is.
+
+Otherwise if the platform method result `items` schema has a property that matches the `x-response` schema on the provider method then each provider value **MUST** be composed into an object under the corresponding property name and the platform **MUST** apply any [result transformations](#9-result-transformations) to the composed result.
+
+### 4.4. Pass-through notifications
+Firebolt events have a synchronous subscriber registration method, e.g. `Lifecycle.onInactive(true)`, in addition to asynchronous notifications when the event actually happens. For events powered by an app pass-through, only the asynchronous notifications are passed in by the providing app. The initial event registration is handled by the platform, and the success response is not handled by the providing app.
+
+This section only applies to platform methods that have an `event` tag.
+
+App provided event registration **MUST** not return an availability error due to a lack of providers, since one may be launched at a future point.
+
+**TODO**: do we want to search the catalog and return unavailable if none are possible? Seems overkill.
+
+To ensure that event provider methods all behave the same the provider method **MUST** have a `result` schema with `"type"` set to `"null"`, since it will not expect any data in the response from the platform after pushing the notification.
+
+The platform method result schema **MUST** either:
+
+> Match the *last* parameter schema on the provider method so that the result can be passed through.
+>
+> Have a property that matches the *last* parameter schema on the provider method so that the result can be passed through.
+
+**TODO**: Find event context params spec and move into this branch
+
+Example platform method with context:
+```json
+{
+    "name": "onFoo",
+    "tags": [
+        {
+            "name": "capabilities",
+            "x-uses": [
+                "xrn:firebolt:capabilities:example:foo"
+            ],
+            "x-provided-by": "foo"
+        },
+        {
+            "name": "event"
+        }
+    ],
+    "params": [
+        {
+            "name": "context1",
+            "schema":{
+                "type": "string"
+            }
+        },
+        {
+            "name": "context2",
+            "schema": {
+                "type": "number"
+            }
+        }
+    ],
+    "result": {
+        "name": "value",
+        "schema": {
+            "type": "boolean"
+        }
+    }
+}
+```
+
+Matching provider method:
+
+```json
+{
+    "name": "foo",
+    "tags": [
+        {
+            "name": "capabilities",
+            "x-provides": "xrn:firebolt:capabilities:example:foo"
+        },
+        {
+            "name": "event"
```

Comment:
The provider method wouldn't be an `event` right?

Reply:
good catch